src/colorizer/Dataset.ts

```diff
@@ -573,7 +580,9 @@ export default class Dataset {
       }, [] as number[]);
     }
 
-    return new Track(trackId, times, ids, centroids, bounds);
+    const track = new Track(trackId, times, ids, centroids, bounds);
+    this.cachedTracks.set(trackId, track);
+    return track;
```

Comment:
this function could now be called getTrack since it doesn't always have to build one